node/core/approval-voting/src/lib.rs

```diff
@@ -752,10 +872,21 @@ async fn handle_actions(
 			Action::WriteCandidateEntry(candidate_hash, candidate_entry) => {
 				transaction.put_candidate_entry(candidate_hash, candidate_entry.into());
 			}
+			Action::IssueApproval(approval_request) => {
+					let mut sender = ctx.sender().clone();
+					let next_actions: Vec<Action> = issue_approval(
+						&mut sender,
+						state,
+						metrics,
+						approval_request,
+					)?.into_iter().map(|v| v.clone()).chain(actions_iter).collect();
+					actions_iter = next_actions.into_iter();
```

Comment:
This seems correct but could use some comprehensive comments about correctness for future readers missing our context. Link to #3311 would be good too.